### ISSUE TITLE
Version 0.14.8 NoMethodError on StreamingOperationalLimits requires newer Google API dependency

### ISSUE THREAD

**Issue by TJC** [spotify/scio#5513]
After upgrading from Scio 0.14.6 to 0.14.8, our Google Dataflow pipeline immediately errored out, with:
```
java.lang.NoSuchMethodError: 'com.google.api.services.dataflow.model.StreamingOperationalLimits com.google.api.services.dataflow.model.StreamingConfigTask.getOperationalLimits()'
```

I was able to fix this by adding an explicit dependency in sbt: 
`"com.google.apis" % "google-api-services-dataflow" % "v1b3-rev20240817-2.0.0"`

It appears that the version Scio specifies (v1b3-rev20240624-2.0.0) is too old to include the above method, but it is required by Dataflow.

**Comment by RustedBones** [comment]
Checking upstream in beam, `v1b3-rev20240817-2.0.0` will be used for future release (2.60). Current release are still using `v1b3-rev20240817-2.0.0`

**Comment by TJC** [comment]
Thanks for investigating this so quickly.

> This dependency is however part of scio's Provided scope, and should be pulled by the runner implementation (in your case beam-runners-google-cloud-dataflow-java).
> Was the google-api-services-dataflow library part of your classpath and uploaded to the staging bucket?

We are pulling in `beam-runners-google-cloud-dataflow-java`, and I've tried with both version 2.57 and 2.59.
I can confirm that the classpath included `google-api-services-dataflow...jar`.

With Scio 0.14.6 it worked fine, but something about 0.14.8 changed - I'm not sure what. Rolling back to 0.14.6 made the issue disappear again, and no other changes were made between.

**Comment by TJC** [comment]
The project is fairly small and simple, and didn't change between Scio versions. The app is packaged with the sbt-pack plugin - this essentially just creates `target/pack/` directory, where `target/pack/lib/` is the classpath containing all the jars required.

Looking at the contents of that directory:
- With Scio 0.14.6, BEAM Runner 2.57: google-api-services-dataflow-v1b3-rev20240218-2.0.0.jar
- With Scio 0.14.8, BEAM Runner 2.57: google-api-services-dataflow-v1b3-rev20240218-2.0.0.jar (too old)
- With Scio 0.14.8, BEAM Runner 2.59: google-api-services-dataflow-v1b3-rev20240624-2.0.0.jar (contains class, but we're seeing error)
- With Scio 0.14.8, BEAM Runner 2.59 and explicit require of g.a.s.d: google-api-services-dataflow-v1b3-rev20240817-2.0.0.jar (contains class, no errors for us)

I'm confused about what could be causing the issue, but if you're sure it's at our end I am happy to take that and try to figure things out. If others hit the issue I guess they'll be able to find and comment on this issue, open or closed.

**Comment by RustedBones** [comment]
Thanks for the investigation.

You should use the beam runner matching with scio's dependency. The combination _Scio 0.14.8, BEAM Runner 2.57_ has great chance to fail as this version depends on 2.59.

If you've added `beam-runners-google-cloud-dataflow-java` as `Runtime` dependency, make sure you use the proper scoping with `Runtime/pack`.

Can you also check if somehow, there is another version of the `StreamingConfigTask` class in your classpath ?
```sh
target/pack/lib> grep -i StreamingConfigTask *
grep: com.google.apis.google-api-services-dataflow-v1b3-rev20240624-2.0.0.jar: binary file matches
```
If you get more entries, that can mean you have a rogue dependency

**Comment by TJC** [comment]
> You should use the beam runner matching with scio's dependency. The combination _Scio 0.14.8, BEAM Runner 2.57_ has great chance to fail as this version depends on 2.59.

Indeed, I would expect that too. I just included it in the matrix of results above to be complete, in case it highlighted any other issue.

> Can you also check if somehow, there is another version of the `StreamingConfigTask` class in your classpath ?
> If you get more entries, that can mean you have a rogue dependency

Thanks for suggesting that. However no other files apart from `google-api-services-dataflow-v1b3-rev20240624-2.0.0.jar` show up with that class. Our dataflow pipeline is relatively simple and doesn't have a huge number of dependencies.

The strange thing here is that everything was working fine prior to the Scio update, but it sounds like you don't know of anything that changed at Scio's end that could have caused it.

Meanwhile -- I've noticed that we were still using a fairly old version of Google's cloud sdk docker image in the build and deploy pipeline; updating to the latest one has also bumped the JDK from 11 to 17. This version appears to be working OK in staging without the above errors occurring.
